docs/build/advanced-concepts/metadata-views.md

```diff
@@ -0,0 +1,226 @@
+```
+
+### Royalties Metadata
+
+Royalty information is vital for the sustainable economics of the creators in the NFT space. The `Royalties` metadata view spells out the specifics of any royalty agreements in place, detailing the percentage of sales revenue that will go to the original creator or other stakeholders on secondary sales.
+
+```cadence
+case Type<MetadataViews.Royalties>():
+    // Assuming each 'Royalty' in the 'royalties' array has 'cut' and 'description' fields
+    let detailedRoyalties = self.royalties.map { royalty ->
+        MetadataViews.Royalty(
+            receiver: royalty.receiver, // The beneficiary of the royalty
+            cut: royalty.cut,          // The percentage cut of each sale
+            description: royalty.description // A description of the royalty terms
+        )
+    }
+    return MetadataViews.Royalties(detailedRoyalties)
+```
+
+### External URL Metadata
+
+The ExternalURL view directs to an associated webpage, providing additional content or information about the NFT.
+
+```cadence
+case Type<MetadataViews.ExternalURL>():
+    return MetadataViews.ExternalURL("<https://example-nft.onflow.org/>".concat(self.id.toString()))
+```
+
+### Traits Metadata
+
+Traits in NFT metadata encapsulate the unique attributes of an NFT, like its visual aspects or any other category-defining properties. These can be essential for marketplaces that need to sort or filter NFTs based on these characteristics.
+
+The `dictToTraits` function in Cadence is designed to convert a dictionary of metadata into a standard array of `Trait` structures. This function helps in preparing the traits data for the `MetadataViews.Traits` view, which can be understood by platforms within the Flow ecosystem.
+
+Here's a more detailed look at how you might implement the `dictToTraits` function and the `Traits` view:
+
+```cadence
+case Type<MetadataViews.Traits>():
+    // Exclude certain traits from being displayed
+    let excludedTraits = ["mintedTime", "foo"]
+
+    // Convert the remaining metadata into an array of `Trait` objects
+    let traitsArray = MetadataViews.dictToTraits(dict: self.metadata, excludedNames: excludedTraits)
+
+    return traitsArray
+```
+
+## Contract-Level Metadata Implementation
+
+Contract-level metadata provides a holistic view of an NFT collection, capturing overarching attributes and contextual information about the entire set, rather than specifics of individual tokens. These views describe attributes at the collection or series level rather than individual NFTs.
+
+### NFTCollectionData
+
+This view provides paths and types related to the NFT collection's storage and access within the smart contract.
+
+```cadence
+case Type<MetadataViews.NFTCollectionData>():
+    return MetadataViews.NFTCollectionData(
+        storagePath: ExampleNFT.CollectionStoragePath,
+        publicPath: ExampleNFT.CollectionPublicPath,
+        providerPath: /private/exampleNFTCollection,
```

Comment:
this should be a string, no?

Reply:
no it is a path, put private paths are going away so should we create a new Collection view?

Reply:
I mean we do not need type anymore after stable cadence